packages/headless/src/features/query-suggest/query-suggest-slice.ts

```diff
@@ -90,8 +92,27 @@ export const querySuggestReducer = createReducer(
 
         querySuggest.q = expression;
       })
+      .addCase(restoreSearchParameters, (state, action) => {
+        updateAllQuerySuggestSetQuery(state, action.payload.q ?? '');
+      })
+      .addCase(executeSearch.fulfilled, (state, action) => {
+        const {queryExecuted} = action.payload;
+        updateAllQuerySuggestSetQuery(state, queryExecuted);
+      })
 );
 
+function updateQuerySuggestQuery(state: QuerySuggestState, query: string) {
+  state.q = query;
+  state.completions = [];
+  state.partialQueries = [];
```

Comment:
Should `partialQueries` be reset every time the query changes? Or should it happen only when a query suggestion is selected?

Reply:
I think it should... wasn't sure here either, I'll make the changes!